esmf_regrid/schemes.py

```diff
@@ -769,17 +769,83 @@ def _regrid_rectilinear_to_unstructured__perform(src_cube, regrid_info, mdtol):
 
 
 def _regrid_unstructured_to_unstructured__prepare(
-    src_grid_cube,
-    target_mesh_cube,
+    src_mesh_cube,
+    tgt_cube_or_mesh,
     method,
     precomputed_weights=None,
+    src_mask=None,
+    tgt_mask=None,
+    src_location=None,
     tgt_location=None,
 ):
```

Comment:
It may be worth adding a short docstring to this similar to `unstructured_to_rectilinear`. It is also worth adding a docstring to the equivalent `rectilinear_to_rectilinear` functions for the sake of consistency.

It may also be worth rethinking which functions these docstrings refer to since they are now more likely to be called by a regridding scheme than by a standalone function. That may be out of scope for this PR though